administrator/components/com_media/views/media/tmpl/default.php

```diff
@@ -43,6 +43,8 @@
 	'editViewUrl'             => JUri::root() . 'administrator/index.php?option=com_media&view=file' . $tmpl,
 	'allowedUploadExtensions' => $params->get('upload_extensions', ''),
 	'maxUploadSizeMb'         => $params->get('upload_maxsize', 10),
+	'defaultFileSystem'       => 'local',
```

Comment:
This is hardcoded and should be removed. You can a "is_default: true" to the providers array.